pycuda_cudf_integration/README.md

```diff
@@ -0,0 +1,113 @@
+
+## Working with external CUDA libraries
+To showcase how this can be done, we will be using a simple example of how we can fill a `cudf` column with random numbers.
+
+```python
+import pycuda.autoprimaryctx
+from pycuda.compiler import SourceModule
+import cudf
+import cupy as cp
+
+df = cudf.DataFrame({'col': cp.zeros(200000)})
+length = cp.int32(len(df['col']))
+```
+We can construct the sample kernel like below,
+```python
+mod = SourceModule('''
+#include <thrust/random.h>
+
+extern "C" {
+
+    __global__ void random_column(double* a, int N)
+    {
+      thrust::default_random_engine rng;
+      
+      int stride = blockDim.x * gridDim.x;
+      for (int i = blockIdx.x * blockDim.x + threadIdx.x; i < N; i += stride) {
+        if (i < N) {
+          rng.discard(i);
+          thrust::uniform_real_distribution<float> rand01(0,1);
+          double r = rand01(rng);
+          a[i] = r;
+        }
+      }
+    }
+}
+''', no_extern_c=True)
+
+func = mod.get_function('random_column')
+func(df['col'], length, block=(256,1,1), grid=(4096,))
+```
+Here we use `thrust` to generate random numbers to fill in the cudf. Below is another interesting column operation that reverses the contents of a column.
+```python
+import pycuda.autoprimaryctx
+from pycuda.compiler import SourceModule
+import cudf
+import cupy as cp
+
+df = cudf.DataFrame({'col': [i for i in range(100)]})
+length = cp.int32(len(df['col'])) - 1
+
+mod = SourceModule('''
+__global__ void reverse_row(long* reverse_row, int N)
+{      
+  int stride = blockDim.x * gridDim.x;
+  for (int i = blockIdx.x * blockDim.x + threadIdx.x; i < N; i += stride) {
+    if (i < N - i) {
+        int row1 = reverse_row[i];
+        int row2 = reverse_row[N - i];
+        reverse_row[i] = row2;
+        reverse_row[N - i] = row1;
+    }
+  }
+}
+''')
+
+func = mod.get_function('reverse_row')
+func(df['col'], length, block=(256,1,1), grid=(4096,))
+```
+
+## When to use what?
+After reading the above, a question that may come to mind is when should you use `Pycuda`, `Cython` or any other alternatives that allow interfacing between `Python` and low-level `C` code? `Pycuda` is likely best used when the developer is looking for an easy and low overhead way of interfacing with a `CUDA` kernel for the purpose of accelerating an operation on their fixed length column data, as `Pycuda` allows the developer to access the power of GPU with little overhead. On the other hand, if the developer is okay with a large amount of `Cython` overhead and requires the ability to execute host-level code, than building `Cython` bindings will likely be the more appropriate option.
```

Comment:
```suggestion
After reading the above, a question that may come to mind is when should you use `Pycuda`, `Cython` or any other alternatives that allow interfacing between `Python` and low-level `C` code? `Pycuda` is likely a good fit when the developer is looking for an easy and low overhead way of interfacing with a `CUDA` kernel for the purpose of accelerating an operation on their fixed length column data. `Pycuda` allows the developer to access the power of GPU with little overhead. On the other hand, if the developer is okay with a large amount of `Cython` overhead and requires the ability to execute host-level code, than building `Cython` bindings will likely be the more appropriate option.
```

Generally good to avoid terms like "best" when there are tradeoffs. Also, fixed width vs fixed length column (same question as above).